### ISSUE TITLE
Data discrepancy issue

### ISSUE THREAD

**Issue by hafizrahman** [woocommerce/woocommerce-ios#12231]
Testing the latest trunk (up to https://github.com/woocommerce/woocommerce-ios/commit/6281cba7a28e304ab0ab1bb3bd4613392b1f08db), I'm seeing this issue.

To replicate:
1. Build app,
2. Create custom range (mine is Feb 1, 2020 to March 7, 2024),
3. Wait until range is loaded.
4. Select a year value in the graph.

Issue:

Looking at the API response, the /stats/visits request return some values for 2023 and 2024:

```
{
  "date": "2024-03-07",
  "unit": "year",
  "fields": [
    "period",
    "visitors",
    "views"
  ],
  "data": [
    [
      "2005-01-01",
      0,
      0
    ],
    [
      "2006-01-01",
      0,
      0
    ],
    [
      "2007-01-01",
      0,
      0
    ],
    [
      "2008-01-01",
      0,
      0
    ],
    [
      "2009-01-01",
      0,
      0
    ],
    [
      "2010-01-01",
      0,
      0
    ],
    [
      "2011-01-01",
      0,
      0
    ],
    [
      "2012-01-01",
      0,
      0
    ],
    [
      "2013-01-01",
      0,
      0
    ],
    [
      "2014-01-01",
      0,
      0
    ],
    [
      "2015-01-01",
      0,
      0
    ],
    [
      "2016-01-01",
      0,
      0
    ],
    [
      "2017-01-01",
      0,
      0
    ],
    [
      "2018-01-01",
      0,
      0
    ],
    [
      "2019-01-01",
      0,
      0
    ],
    [
      "2020-01-01",
      0,
      0
    ],
    [
      "2021-01-01",
      0,
      0
    ],
    [
      "2022-01-01",
      0,
      0
    ],
    [
      "2023-01-01",
      14,
      17
    ],
    [
      "2024-01-01",
      1,
      9
    ]
  ]
}
```

However, in the app, these numbers are not reflected and showing zero instead:

<img src="https://github.com/woocommerce/woocommerce-ios/assets/266376/e610c2f3-d582-40cf-82b6-4f413a771130" width="42%" />

**Comment by dangermattic** [comment]
🚫 Please add a type label (e.g. **type: enhancement**) and a feature label (e.g. **feature: stats**) to this issue.

**Comment by itsmeichigo** [comment]
Closing as this has been resolved by #12232.
